src/hooks/useEvidences.js

```diff
@@ -14,34 +12,31 @@ export default function useEvidences(evidences) {
         return
       }
 
-      setEvidenceProcessed(prev =>
-        [
-          ...prev,
-          evidence,
-        ].sort(({ createdAt: dateLeft }, { createdAt: dateRight }) =>
-          dayjs(dateLeft).isAfter(dayjs(dateRight))
-            ? 1
-            : dayjs(dateLeft).isSame(dayjs(dateRight))
-            ? 0
-            : -1
+      setEvidenceProcessed(prev => {
+        if (prev.some(element => element.id === evidence.id)) {
```

Comment:
I'm wondering if we shouldn't add a mapping that will act as a global cache for files fetched from IPFS into this hook instead of doing this deduping here (e.g. an example cache that only "saves" on success: [`AsyncRequestCache`](https://github.com/aragon/aragon.js/blob/master/packages/aragon-wrapper/src/utils/AsyncRequestCache.js)).

Doing this here is expensive, because we've already had to re-fetch the file, only to realize we should throw it away.